addons/account/static/src/helpers/account_tax.js

```diff
@@ -409,4 +431,488 @@ export const accountTaxHelpers = {
+            amounts.raw_base_amount_currency = roundPrecision(amounts.raw_base_amount_currency, amounts.currency_pd);
+            amounts.raw_base_amount = roundPrecision(amounts.raw_base_amount, amounts.company_currency_pd);
+        }
+
+        // Dispatch the delta across the base lines.
+        for (const amounts of Object.values(total_per_tax)) {
+            if (!amounts.base_lines.length){
+                continue;
+            }
+
+            const base_line = amounts.base_lines.sort(
+                (a, b) => a.tax_details.total_included_currency - b.tax_details.total_included_currency
+            )[0];
+
+            const tax_details = base_line.tax_details;
+            const [index, tax_data] = tax_details.taxes_data.map((x, i) => [i, x]).find(([i, x]) => x.tax.id === amounts.tax.id);
+
+            const delta_base_amount_currency = amounts.raw_base_amount_currency - amounts.base_amount_currency;
+            const delta_base_amount = amounts.raw_base_amount - amounts.base_amount;
+
+            if (index === 0) {
+                tax_details.delta_base_amount_currency += delta_base_amount_currency;
+                tax_details.delta_base_amount += delta_base_amount;
+            }
+
+            tax_data.base_amount_currency += delta_base_amount_currency;
+            tax_data.base_amount += delta_base_amount;
+            tax_data.tax_amount_currency += amounts.raw_tax_amount_currency - amounts.tax_amount_currency;
+            tax_data.tax_amount += amounts.raw_tax_amount - amounts.tax_amount;
+        }
+    },
+
+    // -------------------------------------------------------------------------
+    // TAX TOTALS SUMMARY
+    // -------------------------------------------------------------------------
+
+    get_tax_totals_summary(base_lines, currency, company, {cash_rounding = null} = {}) {
+        const company_pd = company.currency_id.rounding;
+        const tax_totals_summary = {
+            currency_id: currency.id,
+            currency_pd: currency.rounding,
+            company_currency_id: company.currency_id.id,
+            company_currency_pd: company.currency_id.rounding,
+            has_tax_groups: false,
+            subtotals: [],
+            base_amount_currency: 0.0,
+            base_amount: 0.0,
+            tax_amount_currency: 0.0,
+            tax_amount: 0.0,
+        };
+
+        // Global tax values.
+        const global_grouping_function = (base_line, tax_data) => true;
+
+        let base_lines_aggregated_values = this.aggregate_base_lines_tax_details(base_lines, global_grouping_function);
+        let values_per_grouping_key = this.aggregate_base_lines_aggregated_values(base_lines_aggregated_values);
+
+        for (const values of Object.values(values_per_grouping_key)) {
+            if (values.grouping_key) {
+                tax_totals_summary.has_tax_groups = true;
+            }
```

Comment:
It doesn't work because I want to set it true when there is at least one grouping_key. If I do your modification, if the last grouping_key is falsy, I could have a `has_tax_groups=false` even if I have a not falsy grouping_key